src/python/gudhi/tensorflow/cubical_layer.py

```diff
@@ -0,0 +1,72 @@
+import numpy               as np
+import tensorflow          as tf
+from ..cubical_complex  import CubicalComplex
+
+######################
+# Cubical filtration #
+######################
+
+# The parameters of the model are the pixel values.
+
+def _Cubical(X, dimensions):
+    # Parameters: X (image),
+    #             dimensions (homology dimensions)
+
+    # Compute the persistence pairs with Gudhi
+    Xs = X.shape
+    cc = CubicalComplex(dimensions=Xs, top_dimensional_cells=X.flatten())
+    cc.persistence()
+
+    L_cofs = []
+    for dim in dimensions:
+
+        try:
+            cof = cc.cofaces_of_persistence_pairs()[0][dim]
```

Comment:
The call to `cofaces_of_persistence_pairs` is inside the loop on dimensions? Why not make it just once before the loop?

Reply:
You are right, I put it outside the loop.